### ISSUE TITLE
[api v2] operation for job

### ISSUE THREAD

**Issue by ColdWaterLW** [actiontech/dtle#678]
add apis:
- [x] resume job -> /v2/job/resume [post]
- [x] pause job -> /v2/job/pause [post]
- [x] delete job -> /v2/job/delete [post]
       delete job from nomad and delete its metadata from consul

related to #605

**Comment by asiroliu** [comment]
version: 3.21.05.0-3.21.05.x-ccd01e5
1. 创建job
2. 暂停job /v2/job/pause [post]
3. 源端插入数据
4. 恢复job /v2/job/resume [post]
5. 数据不一致，目标端少数据

**Comment by asiroliu** [comment]
# 验证 /v2/job/delete
# version:
```
3.21.05.0-3.21.05.x-ccd01e5
```
# reproduce:
1. deolpy dtle cluster
4. create job
```
shell> cat test.json
{
  "job_name": "test",
  "failover": true,
  "src_task": {
    "task_name": "src",
    "mysql_connection_config": {
      "mysql_host": "172.100.9.1",
      "mysql_password": "test",
      "mysql_port": 3306,
      "mysql_user": "test"
    },
    "replicate_do_db": [
      {
        "table_schema": "test"
      }
    ]
  },
  "dest_task": {
    "task_name": "dest",
    "mysql_connection_config": {
      "mysql_host": "172.100.9.2",
      "mysql_password": "test",
      "mysql_port": 3306,
      "mysql_user": "test"
    }
  }
}
```
3. create migration job
```
shell> curl -s -X POST "http://172.100.9.11:8190/v2/job/migration" -H "Content-Type: application/json" -d @test.json | jq
{
  "job": {
    "job_name": "test",
    "job_id": "test-migration",
    "failover": true,
    "is_mysql_password_encrypted": false,
    "src_task": {
      "task_name": "src",
      "gtid": "",
      "group_max_size": 1,
      "chunk_size": 2000,
      "drop_table_if_exists": false,
      "skip_create_db_table": false,
      "repl_chan_buffer_size": 32,
      "replicate_do_db": [
        {
          "table_schema": "test",
          "table_schema_rename": null,
          "tables": null
        }
      ],
      "replicate_ignore_db": null,
      "mysql_connection_config": {
        "mysql_host": "172.100.9.1",
        "mysql_port": 3306,
        "mysql_user": "test",
        "mysql_password": "*"
      }
    },
    "dest_task": {
      "task_name": "dest",
      "parallel_workers": 1,
      "mysql_connection_config": {
        "mysql_host": "172.100.9.2",
        "mysql_port": 3306,
        "mysql_user": "test",
        "mysql_password": "*"
      }
    }
  },
  "eval_create_index": 13,
  "job_modify_index": 12,
  "message": "ok"
}
```
4. check job list
```
shell> curl -s -X GET "http://172.100.9.11:8190/v2/jobs" | jq
{
  "jobs": [
    {
      "job_id": "test-migration",
      "job_name": "test",
      "job_status": "running",
      "job_status_description": ""
    }
  ],
  "message": "ok"
}
```
5. check cousul matedata
```
shell> curl -s -X GET "http://172.100.9.11:8500/v1/kv/dtle/test?recurse" | jq
[
  {
    "LockIndex": 0,
    "Key": "dtle/test/BinlogFilePos",
    "Flags": 3304740253564472300,
    "Value": "bXlzcWwtYmluLjAwMDAwMi8vZHRsZS8vMTY0MA==",
    "CreateIndex": 32,
    "ModifyIndex": 32
  },
  {
    "LockIndex": 0,
    "Key": "dtle/test/Gtid",
    "Flags": 3304740253564472300,
    "Value": "MDAwMDMzMDYtMDAwMC0wMDAwLTAwMDAtMDAwMDAwMDAzMzA2OjEtMTYsYzJlNmE3MjEtY2E3Ni0xMWViLTljYzYtMDI0MmFjNjQwOTAxOjEtNg==",
    "CreateIndex": 31,
    "ModifyIndex": 31
  },
  {
    "LockIndex": 0,
    "Key": "dtle/test/NatsAddr",
    "Flags": 3304740253564472300,
    "Value": "MTcyLjEwMC45LjExOjgxOTM=",
    "CreateIndex": 28,
    "ModifyIndex": 30
  },
  {
    "LockIndex": 0,
    "Key": "dtle/test/ReplChanBufferSize",
    "Flags": 3304740253564472300,
    "Value": "MzI=",
    "CreateIndex": 27,
    "ModifyIndex": 27
  }
]
```
6. delete job
```
shell> curl -s -X POST "http://10.186.16.109:18190/v2/job/delete" -H "accept: application/json" -H "Content-Type: application/x-www-form-urlencoded" -d "job_id=test-migration" | jq
{
  "message": "ok"
}
```
7. check job list
```
shell> curl -s -X GET "http://172.100.9.11:8190/v2/jobs" | jq
{
  "jobs": [],
  "message": "ok"
}
```
8. check cousul matedata
```
shell> curl -s -X GET "http://172.100.9.11:8500/v1/kv/dtle/test?recurse" | jq
```

**Comment by ffffwh** [comment]
## case 2: job stuck after several TX

- some chan has not been restarted
- should re-create the `Extractor/Applier` instead of using the previous one.

**Comment by asiroliu** [comment]
# 验证
# version:
```
3.21.05.0-3.21.05.x-4acaef3
```
# reproduce:
1. deolpy dtle cluster
2. create job
```
shell> cat test.json
{
  "job_id": "test",
  "failover": true,
  "src_task": {
    "task_name": "src",
    "node_id": "a09ee01f-bae6-fc9c-e0ae-dbbbc8253863",
    "mysql_connection_config": {
      "mysql_host": "172.100.9.1",
      "mysql_password": "test_src",
      "mysql_port": 3306,
      "mysql_user": "test_src"
    },
    "replicate_do_db": [
      {
        "table_schema": "test"
      }
    ]
  },
  "dest_task": {
    "task_name": "dest",
    "node_id": "0fbcce0c-4274-cafc-c1cb-281d3bb8f330",
    "mysql_connection_config": {
      "mysql_host": "172.100.9.2",
      "mysql_password": "test_dest",
      "mysql_port": 3306,
      "mysql_user": "test_dest"
    }
  }
}
```
3. create migration job
```
shell> curl -s -X POST "http://172.100.9.11:8190/v2/job/migration" -H "Content-Type: application/json" -d @test.json | jq
{
  "job": {
    "job_id": "test-migration",
    "task_step_name": "",
    "failover": true,
    "is_mysql_password_encrypted": false,
    "src_task": {
      "task_name": "src",
      "node_id": "a09ee01f-bae6-fc9c-e0ae-dbbbc8253863",
      "gtid": "",
      "group_max_size": 1,
      "chunk_size": 2000,
      "drop_table_if_exists": false,
      "skip_create_db_table": false,
      "repl_chan_buffer_size": 32,
      "replicate_do_db": [
        {
          "table_schema": "test",
          "table_schema_rename": null,
          "tables": null
        }
      ],
      "replicate_ignore_db": null,
      "mysql_connection_config": {
        "mysql_host": "172.100.9.1",
        "mysql_port": 3306,
        "mysql_user": "test_src",
        "mysql_password": "*"
      }
    },
    "dest_task": {
      "task_name": "dest",
      "node_id": "0fbcce0c-4274-cafc-c1cb-281d3bb8f330",
      "parallel_workers": 1,
      "mysql_connection_config": {
        "mysql_host": "172.100.9.2",
        "mysql_port": 3306,
        "mysql_user": "test_dest",
        "mysql_password": "*"
      }
    }
  },
  "eval_create_index": 16,
  "job_modify_index": 15,
  "message": "ok"
}
```
4. insert data
```
shell> sysbench /usr/share/sysbench/oltp_insert.lua --mysql-host=172.100.9.1 --mysql-port=3306 --mysql-user=test --mysql-password=test --create_secondary=off --report-interval=10 --mysql-db=test --tables=2 --table_size=1000 prepare
sysbench 1.0.20 (using bundled LuaJIT 2.1.0-beta2)

Creating table 'sbtest1'...
Inserting 1000 records into 'sbtest1'
Creating table 'sbtest2'...
Inserting 1000 records into 'sbtest2'
```
5. check data consistency
```
shell> mysqldbcompare --server1=test:test@172.100.9.1:3306 --server2=test:test@172.100.9.2:3306 test --disable-binary-logging
# WARNING: Using a password on the command line interface can be insecure.
# server1 on 172.100.9.1: ... connected.
# server2 on 172.100.9.2: ... connected.
# Checking databases test on server1 and test on server2
#
#                                                   Defn    Row     Data
# Type      Object Name                             Diff    Count   Check
# -------------------------------------------------------------------------
# TABLE     sbtest1                                 pass    pass    -
#           - Compare table checksum                                pass
# TABLE     sbtest2                                 pass    pass    -
#           - Compare table checksum                                pass

# Databases are consistent.
#
# ...done
```
6. pause job
```
shell> curl -s -X POST "http://172.100.9.11:8190/v2/job/pause" -H "accept: application/json" -H "Content-Type: application/x-www-form-urlencoded" -d "job_id=test-migration" | jq
{
  "message": "ok"
}
```
7. check log
```
# src dtle log
2021-06-22T16:38:22.116+0800 [DEBUG] client.driver_mgr.dtle: TimestampContext.GetDelay: driver=dtle delay=0 job=test-migration @module=dtle.extractor timestamp=2021-06-22T16:38:22.116+0800
2021-06-22T16:38:24.650+0800 [DEBUG] client.driver_mgr.dtle: SignalTask: driver=dtle @module=dtle signal=pause taskID=dfcda4c4-c690-ef45-d77a-3811545fa04f/src/699d3955 timestamp=2021-06-22T16:38:24.650+0800
2021-06-22T16:38:24.650+0800 [INFO]  client.driver_mgr.dtle: pause a task: driver=dtle @module=dtle taskID=dfcda4c4-c690-ef45-d77a-3811545fa04f/src/699d3955 timestamp=2021-06-22T16:38:24.650+0800
2021-06-22T16:38:24.650+0800 [DEBUG] client.driver_mgr.dtle: extractor shutdown: driver=dtle @module=dtle.extractor job=test-migration timestamp=2021-06-22T16:38:24.650+0800
2021-06-22T16:38:24.650+0800 [INFO]  client.driver_mgr.dtle: extractor shutdown: driver=dtle @module=dtle.extractor job=test-migration timestamp=2021-06-22T16:38:24.650+0800
2021-06-22T16:38:24.650+0800 [DEBUG] client.driver_mgr.dtle: BinlogReader.Close: driver=dtle @module=reader job=test-migration timestamp=2021-06-22T16:38:24.650+0800
2021-06-22T16:38:24.651+0800 [ERROR] client.driver_mgr.dtle: error GetEvent.: driver=dtle job=test-migration @module=reader err="Sync was closed" timestamp=2021-06-22T16:38:24.651+0800
2021-06-22T16:38:24.655+0800 [DEBUG] client.driver_mgr.dtle: StreamEvents goroutine exited: driver=dtle job=test-migration @module=dtle.extractor timestamp=2021-06-22T16:38:24.655+0800
2021-06-22T16:38:24.655+0800 [INFO]  client.driver_mgr.dtle: Shutting down: driver=dtle job=test-migration @module=dtle.extractor timestamp=2021-06-22T16:38:24.655+0800
2021-06-22T16:38:24.798+0800 [DEBUG] client: updated allocations: index=28 total=1 pulled=0 filtered=1
2021-06-22T16:38:24.798+0800 [DEBUG] client: allocation updates: added=0 removed=0 updated=0 ignored=1
2021-06-22T16:38:24.798+0800 [DEBUG] client: allocation updates applied: added=0 removed=0 updated=0 ignored=1 errors=0
```
```
# dest dtle log
2021-06-22T16:38:22.118+0800 [DEBUG] client.driver_mgr.dtle: Stats: driver=dtle @module=dtle.applier job=test-migration timestamp=2021-06-22T16:38:22.117+0800
2021-06-22T16:38:22.118+0800 [DEBUG] client.driver_mgr.dtle: TimestampContext.GetDelay: driver=dtle @module=dtle.applier delay=0 job=test-migration timestamp=2021-06-22T16:38:22.117+0800
2021-06-22T16:38:22.145+0800 [DEBUG] client.driver_mgr.dtle: no binlogEntry for 10s: driver=dtle @module=dtle.applier job=test-migration timestamp=2021-06-22T16:38:22.145+0800
2021-06-22T16:38:24.659+0800 [DEBUG] client.driver_mgr.dtle: SignalTask: driver=dtle @module=dtle signal=pause taskID=e5791256-ca37-68e7-c570-94a1606a365c/dest/7fb302d7 timestamp=2021-06-22T16:38:24.659+0800
2021-06-22T16:38:24.659+0800 [INFO]  client.driver_mgr.dtle: pause a task: driver=dtle @module=dtle taskID=e5791256-ca37-68e7-c570-94a1606a365c/dest/7fb302d7 timestamp=2021-06-22T16:38:24.659+0800
2021-06-22T16:38:24.660+0800 [INFO]  client.driver_mgr.dtle: Shutting down: driver=dtle @module=dtle.applier job=test-migration timestamp=2021-06-22T16:38:24.659+0800
2021-06-22T16:38:24.815+0800 [DEBUG] client: updated allocations: index=29 total=1 pulled=0 filtered=1
2021-06-22T16:38:24.815+0800 [DEBUG] client: allocation updates: added=0 removed=0 updated=0 ignored=1
2021-06-22T16:38:24.815+0800 [DEBUG] client: allocation updates applied: added=0 removed=0 updated=0 ignored=1 errors=0
2021-06-22T16:38:37.118+0800 [DEBUG] client.driver_mgr.dtle: Stats: driver=dtle @module=dtle.applier job=test-migration timestamp=2021-06-22T16:38:37.118+0800
```

8. insert data
```
shell> sysbench /usr/share/sysbench/oltp_insert.lua --mysql-host=172.100.9.1 --mysql-port=3306 --mysql-user=test --mysql-password=test --create_secondary=off --report-interval=10 --mysql-db=test --tables=2 --events=100 run
sysbench 1.0.20 (using bundled LuaJIT 2.1.0-beta2)

Running the test with following options:
Number of threads: 1
Report intermediate results every 10 second(s)
Initializing random number generator from current time


Initializing worker threads...

Threads started!

SQL statistics:
    queries performed:
        read:                            0
        write:                           100
        other:                           0
        total:                           100
    transactions:                        100    (1999.12 per sec.)
    queries:                             100    (1999.12 per sec.)
    ignored errors:                      0      (0.00 per sec.)
    reconnects:                          0      (0.00 per sec.)

General statistics:
    total time:                          0.0477s
    total number of events:              100

Latency (ms):
         min:                                    0.33
         avg:                                    0.46
         max:                                    1.06
         95th percentile:                        0.70
         sum:                                   46.28

Threads fairness:
    events (avg/stddev):           100.0000/0.00
    execution time (avg/stddev):   0.0463/0.00
```
9. check data consistency
```
shell> mysqldbcompare --server1=test:test@172.100.9.1:3306 --server2=test:test@172.100.9.2:3306 test --disable-binary-logging
# WARNING: Using a password on the command line interface can be insecure.
# server1 on 172.100.9.1: ... connected.
# server2 on 172.100.9.2: ... connected.
# Checking databases test on server1 and test on server2
#
#                                                   Defn    Row     Data
# Type      Object Name                             Diff    Count   Check
# -------------------------------------------------------------------------
# TABLE     sbtest1                                 FAIL    ERROR: The object definitions do not match.
```
10. resume job
```
shell> curl -s -X POST "http://172.100.9.11:8190/v2/job/resume" -H "accept: application/json" -H "Content-Type: application/x-www-form-urlencoded" -d "job_id=test-migration" | jq
{
  "message": "ok"
}
```
11. check log
```
# src dtle log
2021-06-22T16:39:18.812+0800 [DEBUG] client.driver_mgr.dtle: SignalTask: driver=dtle @module=dtle signal=resume taskID=dfcda4c4-c690-ef45-d77a-3811545fa04f/src/699d3955 timestamp=2021-06-22T16:39:18.812+0800
2021-06-22T16:39:18.812+0800 [INFO]  client.driver_mgr.dtle: resume a task: driver=dtle @module=dtle taskID=dfcda4c4-c690-ef45-d77a-3811545fa04f/src/699d3955 timestamp=2021-06-22T16:39:18.812+0800
2021-06-22T16:39:18.812+0800 [INFO]  client.driver_mgr.dtle: NewExtractor: driver=dtle @module=dtle job=test-migration timestamp=2021-06-22T16:39:18.812+0800
2021-06-22T16:39:18.812+0800 [DEBUG] client.driver_mgr.dtle: NewExtractor. after LoadSchemas: driver=dtle @module=dtle timestamp=2021-06-22T16:39:18.812+0800
2021-06-22T16:39:18.813+0800 [DEBUG] client.driver_mgr.dtle: consul put ReplChanBufferSize: driver=dtle @module=dtle.extractor job=test-migration timestamp=2021-06-22T16:39:18.812+0800
2021-06-22T16:39:18.814+0800 [DEBUG] client.driver_mgr.dtle: SrcWatchNats: driver=dtle @module=dtle timestamp=2021-06-22T16:39:18.814+0800
2021-06-22T16:39:18.815+0800 [INFO]  client.driver_mgr.dtle: NatsAddr. got addr before having put wait: driver=dtle addr=172.100.9.13:8193 @module=dtle timestamp=2021-06-22T16:39:18.815+0800
2021-06-22T16:39:18.819+0800 [INFO]  client.driver_mgr.dtle: NatsAddr. got wait: driver=dtle @module=dtle timestamp=2021-06-22T16:39:18.818+0800
2021-06-22T16:39:18.826+0800 [INFO]  client.driver_mgr.dtle: NatsAddr. got waitdst. will put wait: driver=dtle @module=dtle timestamp=2021-06-22T16:39:18.826+0800
2021-06-22T16:39:18.828+0800 [INFO]  client.driver_mgr.dtle: NatsAddr. got wait: driver=dtle @module=dtle timestamp=2021-06-22T16:39:18.828+0800
2021-06-22T16:39:18.830+0800 [INFO]  client.driver_mgr.dtle: NatsAddr. got addr: driver=dtle @module=dtle addr=172.100.9.13:8193 timestamp=2021-06-22T16:39:18.830+0800
2021-06-22T16:39:18.831+0800 [INFO]  client.driver_mgr.dtle: Got gtid from consul: driver=dtle gtid=00003306-0000-0000-0000-000000003306:1-16,c892663c-d334-11eb-a4a5-0242ac640901:1-10 job=test-migration @module=dtle.extractor timestamp=2021-06-22T16:39:18.831+0800
2021-06-22T16:39:18.831+0800 [INFO]  client.driver_mgr.dtle: Use gtid from consul: driver=dtle job=test-migration @module=dtle.extractor gtid=00003306-0000-0000-0000-000000003306:1-16,c892663c-d334-11eb-a4a5-0242ac640901:1-10 timestamp=2021-06-22T16:39:18.831+0800
2021-06-22T16:39:18.831+0800 [INFO]  client.driver_mgr.dtle: Got BinlogFile/Pos from consul: driver=dtle file= job=test-migration pos=0 @module=dtle.extractor timestamp=2021-06-22T16:39:18.831+0800
2021-06-22T16:39:18.832+0800 [INFO]  client.driver_mgr.dtle: Use BinlogFile/Pos from consul: driver=dtle @module=dtle.extractor file=mysql-bin.000002 job=test-migration pos=1640 timestamp=2021-06-22T16:39:18.831+0800
...
```
```
# dest dtle log
2021-06-22T16:39:18.816+0800 [DEBUG] client.driver_mgr.dtle: SignalTask: driver=dtle @module=dtle signal=resume taskID=e5791256-ca37-68e7-c570-94a1606a365c/dest/7fb302d7 timestamp=2021-06-22T16:39:18.816+0800
2021-06-22T16:39:18.816+0800 [INFO]  client.driver_mgr.dtle: resume a task: driver=dtle @module=dtle taskID=e5791256-ca37-68e7-c570-94a1606a365c/dest/7fb302d7 timestamp=2021-06-22T16:39:18.816+0800
2021-06-22T16:39:18.816+0800 [INFO]  client.driver_mgr.dtle: NewApplier: driver=dtle job=test-migration @module=dtle timestamp=2021-06-22T16:39:18.816+0800
2021-06-22T16:39:18.817+0800 [DEBUG] client.driver_mgr.dtle: got ReplChanBufferSize from consul: driver=dtle @module=dtle.applier i=32 job=test-migration timestamp=2021-06-22T16:39:18.817+0800
2021-06-22T16:39:18.818+0800 [DEBUG] client.driver_mgr.dtle: use ReplChanBufferSize from consul: driver=dtle @module=dtle.applier i=32 job=test-migration timestamp=2021-06-22T16:39:18.817+0800
2021-06-22T16:39:18.819+0800 [INFO]  client.driver_mgr.dtle: Got gtid from consul: driver=dtle @module=dtle.applier gtid=00003306-0000-0000-0000-000000003306:1-16,c892663c-d334-11eb-a4a5-0242ac640901:1-10 job=test-migration timestamp=2021-06-22T16:39:18.819+0800
2021-06-22T16:39:18.819+0800 [INFO]  client.driver_mgr.dtle: Use gtid from consul: driver=dtle @module=dtle.applier gtid=00003306-0000-0000-0000-000000003306:1-16,c892663c-d334-11eb-a4a5-0242ac640901:1-10 job=test-migration timestamp=2021-06-22T16:39:18.819+0800
2021-06-22T16:39:18.820+0800 [INFO]  client.driver_mgr.dtle: Got BinlogFile/Pos from consul: driver=dtle file=mysql-bin.000002 job=test-migration pos=1640 @module=dtle.applier timestamp=2021-06-22T16:39:18.820+0800
2021-06-22T16:39:18.821+0800 [INFO]  client.driver_mgr.dtle: Use BinlogFile/Pos from consul: driver=dtle job=test-migration pos=1640 @module=dtle.applier file=mysql-bin.000002 timestamp=2021-06-22T16:39:18.820+0800
...
```
12. check data consistency
```
shell> mysqldbcompare --server1=test:test@172.100.9.1:3306 --server2=test:test@172.100.9.2:3306 test --disable-binary-logging
# WARNING: Using a password on the command line interface can be insecure.
# server1 on 172.100.9.1: ... connected.
# server2 on 172.100.9.2: ... connected.
# Checking databases test on server1 and test on server2
#
#                                                   Defn    Row     Data
# Type      Object Name                             Diff    Count   Check
# -------------------------------------------------------------------------
# TABLE     sbtest1                                 pass    pass    -
#           - Compare table checksum                                pass
# TABLE     sbtest2                                 pass    pass    -
#           - Compare table checksum                                pass

# Databases are consistent.
#
# ...done
```
13. delete job
```
shell> curl -s -X POST "http://172.100.9.11:8190/v2/job/delete" -H "accept: application/json" -H "Content-Type: application/x-www-form-urlencoded" -d "job_id=test-migration" | jq
{
  "message": "ok"
}
```
14. check job is delete
```
shell> curl -s -X GET "http://172.100.9.11:8190/v2/jobs" | jq
{
  "jobs": [],
  "message": "ok"
}

shell> curl -s -X GET "http://172.100.9.11:8500/v1/kv/dtle/test?recurse" | jq
```
